maltoolbox/language/languagegraph.py

```diff
@@ -1119,11 +1660,11 @@ def get_asset_by_name(
         The asset matching the name.
         None if there is no match.
         """
-        for asset in self.assets:
-            if asset.name == asset_name:
-                return asset
+        if asset_name in self.assets:
+            return self.assets[asset_name]
+        else:
+            return None
```

Comment:
This is not needed, it is basically a dict key lookup that you want to do. If you need to default to None (which I don't think should be the case, it should be the user's responsibility to check if the key/asset they are asking for is valid), you can use a `defaultdict`.